File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int>mp;
        for(auto it:s){
            mp[it]++;
        }
        for(auto it:t){
            mp[it]--;
        }
        for (auto pair : mp) {
            if (pair.second != 0) {
                return false;
            }
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> mp = new HashMap<>();
        int l1 = s.length();
        int l2 = t.length();
        if (l1 != l2) {
            return false;
        }
        for (int i = 0; i < l1; i++) {
            char c1 = s.charAt(i);
            char c2 = t.charAt(i);
            mp.put(c1, mp.getOrDefault(c1, 0) + 1);
            mp.put(c2, mp.getOrDefault(c2, 0) - 1);
        }
        for (Integer value : mp.values()) {
            if (value != 0)
                return false;
        }
        return true;
    }
}